src/midi/AkaiAPC40MkII.h

```diff
@@ -21,6 +23,42 @@ class AkaiAPC40MkII : public MidiController
     ValueControl *sceneLaunchButton(int y);
     ValueControl *clipStopButton(int x);
     ValueControl *triStateButton(int x);
+    ValueControl *numberButton(int x);
+    ValueControl *sButton(int x);
+    ValueControl *recButton(int x);
+    ValueControl *bankSelectLeftButton();
+    ValueControl *bankSelectTopButton();
+    ValueControl *bankSelectRightButton();
+    ValueControl *bankSelectBottomButton();
+    ValueControl *shiftButton();
+    ValueControl *bankButton();
+    ValueControl *detailViewButton();
+    ValueControl *clipDevViewButton();
+    ValueControl *devLockButton();
+    ValueControl *devOnOFfButton();
+    ValueControl *bankRightArrowButton();
+    ValueControl *bankLeftArrowButton();
+    ValueControl *deviceRightArrowButton();
+    ValueControl *deviceLeftArrowButton();
+    ValueControl *nudgePlusButton();
+    ValueControl *nudgeMinusButton();
+    ValueControl *userButton();
+    ValueControl *tapTempoButton();
+    ValueControl *metronomeButton();
+    ValueControl *sendsButton();
+    ValueControl *panButton();
+    ValueControl *playButton();
+    ValueControl *recordButton();
+    ValueControl *sessionButton();
+    ValueControl *cueLevelKnob();
+    ValueControl *TempoKnob();
```

Comment:
Wäre zwar ein Haufen arbeit das jetzt nochmal zu ersetzen, aber ich glaube es ist eindeutiger wenn du ein Enum mit den einzelnen werten aufbaust und dann nur noch eine Methode getValueControl( AkaiAPC40MkII::Buttons button) hast. Damit hat man in den einzelnen Methoden keine magic numbers mehr und weniger Methoden.

Reply:
Faulheit - gibt ja 
`ValueControl *MidiController::controlByName(QString name)`
für genau das. Eigentlich sind die sogar überflüssig

Reply:
Musst du wissen ob du die dann drin lässt oder löschst. Toter Code ist ja meistens nicht so geil. Die eizelnen ids in enums oder const expr zu packen kann ja ne Issue werden die man mal macht wenn man sonst nix zu tun hat.

Reply:
ich schmeiß es gerade raus für `ValueControl *getControl(ControlType type, int offset)` mit
`enum ControlType`, das alle magic control offsets enthält.

Reply:
done.